dev/Tiles/entity.h

```diff
@@ -0,0 +1,66 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include "metatile.h"
+
+typedef struct Entity {
+    MetaTile* metatile[32];
+    unsigned char frameRate;
+    unsigned char numFrames;
+    unsigned int currentFrame;
+} Entity;
+
+Entity* Entity_Create(MetaTile* metatile[], unsigned char numFrames) 
+{
+    Entity* entity = malloc(sizeof(struct Entity));
```

Comment:
I gather it's a choice to use dynamic memory (I seem to remember you asking about it on Discord / the forum at some point?) but to me it is a strange decision for SMS games, simply because you really don't see it used very much.

I'm not a big expert on this, but I would say that dynamic memory must be considerably less efficient the smaller the total RAM space is, since in most implementations it rather aggressively reserves blocks of space for new allocations. That said, I guess 8MB is pretty big by historical standards and `malloc` has been around forever, so perhaps it works well enough - it really does depend on how well the folks over at SDCC have implemented the standard library though. If you are managing your own memory at least you know _exactly_ what is where.

That brings me to me second point which is that if you're writing a game you almost certainly know exactly how much memory you need - at any given time and therefore peak, and so there's very little advantage in using dynamic allocation, if any. You have to think about allocing and freeing, you automatically have to deal with pointers absolutely everywhere and you suddenly have memory leaks as a possible failure mode, which you pretty much don't have to worry about if you manage your own memory.

Anyway, that's all I'll say on the matter - it's certainly possible to do what you want to do without dynamic memory.